ui/src/App.tsx

```diff
@@ -32,16 +33,26 @@ export default function App(props: {connection: Connection}) {
 
   return (
     <div className="App">
-      <header className="App__header">
-        <h1 className="retro-font">Goretro</h1>
-        { state.name ? <span>with <strong>{state.name}</strong></span> : <span>Here comes a new challenger!</span> }
-      </header>
+      { headerComponent(state) }
 
       { mainComponent(props.connection, state, dispatch) }
     </div>
   );
 }
 
+function headerComponent(state: types.State) {
+  if (!state.error && !state.name) {
+    return <Banner/>
+  }
+
+  return <header>
+    <div className="App__header">
+      <h1 className="retro-font">Goretro</h1>
+      { state.name ? <span>with <strong>{state.name}</strong></span> : <span>Here comes a new challenger!</span> }
+    </div>
+  </header>
```

Comment:
the logic here is a bit contrived: if we have an error, why does the header care? Why would the "non-shader" header be better in case of error? And assuming the header does not care about the error, you don't need to check that there's a name at line 51...

In my imagination the "shader" logo would stay on as long as the retro would not start, but that's open for discussion.

Reply:
I imagined the header to be only on the initial landing page but the PR is not about deciding where it is or not, and this can be easily changed. 
On the topic of where to put the shader, we may want to consider that my fan goes full-throttle in a few dozens of seconds, so unless we optimize it, I would minimize where it appears. 

On line 44 it's a check of whether or not we're on the landing page. I don't like it much either because it's not explicit but I didn't find something better for now. This seems ok to me at this stage, given that this code will change in the next days.

On line 51, we display the name if there is one. It makes sense to me that the header checks if there's a name or not.